crates/rolldown/src/module_loader/module_loader.rs

```diff
@@ -174,32 +173,25 @@ impl ModuleLoader {
     }
 
     let mut errors = vec![];
-    let mut all_warnings: Vec<BuildError> = Vec::new();
+    let mut all_warnings: Vec<BuildError> = vec![];
 
-    self
-      .intermediate_normal_modules
-      .modules
-      .reserve(user_defined_entries.len() + 1 /* runtime */);
-    self
-      .intermediate_normal_modules
-      .ast_table
-      .reserve(user_defined_entries.len() + 1 /* runtime */);
-
-    // Store the already consider as entry module
-    let mut user_defined_entry_ids = FxHashSet::with_capacity(user_defined_entries.len());
+    let entries_count = user_defined_entries.len() + /* runtime */ 1;
+    self.intermediate_normal_modules.modules.reserve(entries_count);
+    self.intermediate_normal_modules.ast_table.reserve(entries_count);
 
     let mut entry_points = user_defined_entries
       .into_iter()
       .map(|(name, info)| EntryPoint {
         name: Some(name),
-        id: self.try_spawn_new_task(info, true).expect_normal(),
+        id: self.try_spawn_new_task(info, /* is_user_defined_entry */ true).expect_normal(),
         kind: EntryPointKind::UserDefined,
       })
-      .inspect(|e| {
-        user_defined_entry_ids.insert(e.id);
-      })
       .collect::<Vec<_>>();
 
+    // Store the already consider as entry module
+    let user_defined_entry_ids: FxHashSet<NormalModuleId> =
```

Comment:
Instead of inserting it in place, do it in one go to better utilize cpu cache, otherwise `user_defined_entry_ids` may go out of the cache in the loop above.

`user_defined_entries` is generally small, so there won't be much of an impact.

Reply:
`user_defined_entries` could be huge as there use case like making all files as input. https://rollupjs.org/configuration-options/#input:~:text=If%20you%20want%20to%20convert%20a%20set%20of%20files%20to%20another%20format%20while%20maintaining%20the%20file%20structure%20and%20export%20signatures

Reply:
Let me double check

Reply:
Reverted.